Automattic/pocket-casts-android
# Android 14 - Upgrade cast lib

Part of #1569 

## Description
This updates `play-services-cast-framework` from `21.2.0` to the latest `21.4.0.`

Release Notes:
[21.3.0](https://developers.google.com/cast/docs/release-notes#march-28,-2023)
[21.4.0](https://developers.google.com/cast/docs/release-notes#december-4,-2023)

Changes from `21.3.0` are taken to prepare for Android 14 target SDK update:
> [NotificationManager](https://developer.android.com/reference/android/app/NotificationManager) is now used to post Cast media notifications instead of [Foreground Service](https://developer.android.com/guide/components/foreground-services).

Without this, changing `targetSdkVersion` to 34 (Android 14) crashes the app while casting with exception:

```
FATAL EXCEPTION: main (Ask Studio Bot)
                            Process: au.com.shiftyjelly.pocketcasts, PID: 30400
                            java.lang.RuntimeException: Unable to start service com.google.android.gms.cast.framework.media.MediaNotificationService@fad95f2 with Intent { act=com.google.android.gms.cast.framework.action.UPDATE_NOTIFICATION pkg=au.com.shiftyjelly.pocketcasts cmp=au.com.shiftyjelly.pocketcasts/com.google.android.gms.cast.framework.media.MediaNotificationService (has extras) }: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{dd138da 30400:au.com.shiftyjelly.pocketcasts/u0a1429} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
                            	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4859)
                            	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
                            	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2294)
                            	at android.os.Handler.dispatchMessage(Handler.java:106)
                            	at android.os.Looper.loopOnce(Looper.java:205)
                            	at android.os.Looper.loop(Looper.java:294)
                            	at android.app.ActivityThread.main(ActivityThread.java:8194)
```

I also tested the latest version `21.4.0` and it works well too.

## Testing Instructions

Prerequisites: 
- Change `targetSdkVersion` set to 34 in `dependencies.gradle.kts`
- Requires Cast device

1. Install the app
3. Play an episode
4. Open full-screen player
5. Tap more actions at the bottom
6. Tap `Chromecast`
7. ✅  Notice that casting succeeds without any crash
8. Pull down the notification drawer on the phone
9. Toggle play/ pause
10. ✅  Notice that episode playback can be controlled using the media notification

## Checklist
 
#### I have tested any UI changes...
